server/auth/auth.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2020 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissionKeys and
+// limitations under the License.
+//
+
+package auth
+
+import (
+	"encoding/json"
+	"github.com/tikv/pd/pkg/errs"
+	"github.com/tikv/pd/server/kv"
+	"go.etcd.io/etcd/clientv3"
+	"path"
+	"strings"
+	"sync"
+)
+
+// RBACManager is used for the rbac storage, cache, management and enforcing logic.
+type RBACManager struct {
+	roleManager
+}
+
+type roleManager struct {
+	kv    kv.Base
+	lock  *sync.RWMutex
+	roles map[string]*Role
+}
+
+// newRoleManager creates a new roleManager.
+func newRoleManager(kv kv.Base) *roleManager {
+	return &roleManager{kv: kv, lock: new(sync.RWMutex), roles: make(map[string]*Role)}
```

Comment:
no need to new a mutex?

Reply:
This function is only for debugging purposes, so we need to new a mutex to make roleManager work standalone.

Reply:
Hmm...actually I mean the mutex doesn't need to initialize

Reply:
I was thinking about sharing the same mutex between roleManager & userManager (not implemented in this pr yet). The draft is available at https://github.com/PhotonQuantum/pd/blob/rbac-manager/server/auth/auth.go#L46.

Umm... But it seems that there's no need to have a global mutex between roleManager and userManager. I'm removing it. Thanks!